src/Workspaces/Remote/ServiceHub/Services/SolutionEvents/RemoteSolutionEventsAggregationService.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.DocumentHighlighting;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Microsoft.CodeAnalysis.SolutionEvents;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    internal sealed class RemoteSolutionEventsAggregationService : BrokeredServiceBase, IRemoteSolutionEventsAggregationService
```

Comment:
>RemoteSolutionEventsAggregationService [](http://example.com/codeflow?start=26&length=38)

Nit: Consider renaming the containing dir to `SolutionEventsAggregation` to match the service name (w/o Remote- prefix and -Service suffix).

Reply:
gotcha.

Reply:
I personally think this is fine as teh general component is the 'solution events' component.  This is just the remote side of one particular interface of that component.  keeping the component name (i.e. the folder) consistent across all the projects this touches is desirable to me (and matches what we do with our other components).